rs-lib/src/types.rs

```diff
@@ -264,7 +264,12 @@ where
 pub trait NodeTrait<'a>: SpannedExt {
   fn parent(&self) -> Option<Node<'a>>;
   fn children(&self) -> Vec<Node<'a>>;
+
+  // TODO: according to the convention, the name should preferably be `as_node`
+  // https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv
+  #[allow(clippy::wrong_self_convention)]
   fn into_node(&self) -> Node<'a>;
```

Comment:
As commented, the name `into_node` doesn't follow the naming convention. Personally I feel like it should be renamed to `as_node`, it'll be a breaking change though. Any thoughts?